src/libaktualizr/uptane/imagerepository.cc

```diff
@@ -214,25 +209,24 @@ std::shared_ptr<Uptane::Targets> ImageRepository::verifyDelegation(const std::st
 
 bool ImageRepository::targetsExpired() {
   if (targets->isExpired(TimeStamp::Now())) {
-    last_exception = Uptane::ExpiredMetadata(type.toString(), Role::Targets().ToString());
-    return true;
+    throw Uptane::ExpiredMetadata(type.toString(), Role::Targets().ToString());
   }
   return false;
 }
 
-bool ImageRepository::updateMeta(INvStorage& storage, const IMetadataFetcher& fetcher) {
+void ImageRepository::updateMeta(INvStorage& storage, const IMetadataFetcher& fetcher) {
   resetMeta();
 
   if (!updateRoot(storage, fetcher, RepositoryType::Image())) {
-    return false;
+    throw std::runtime_error("");
```

Comment:
Now I see. I think it makes sense to try to decide on some new meaningful exceptions to use here and in similar places. Did you have particular plans or ideas already?

Reply:
It was changed in later commits. Maybe could have squashed some more stuff.

Reply:
No problem, that was my mistake then. I was trying to go through the commits one by one but crossreference the final version before leaving comments. I must've gotten confused somehow.